File: Assets/Scripts/Referee.js
#pragma strict

public var p1 : boolean = false;
public var changed : boolean = false;

function Awake () {
	DontDestroyOnLoad (transform.gameObject);
}

function FixedUpdate(){
	if (Application.loadedLevelName == "Decision" && !changed){
		Change();
		changed = true;
	}
}

function Player1 (){
	p1 = true;
}

function Player2 (){
	p1 = false;
}

function Change(){
	yield WaitForSeconds (2);
	if (p1){
		Application.LoadLevel ("Celebration_P1");
	}else{
		Application.LoadLevel ("Celebration_P2");
	}
	changed = false;
}

File: Assets/Scripts/_Referee.cs
using UnityEngine;
using System.Collections;

public class _Referee : MonoBehaviour {
	public bool p1 = false;
	public bool changed = false;
	
	public GameObject p1Background;
	public GameObject p2Background;

	void Awake() {
		DontDestroyOnLoad(transform.gameObject);
	}
	
	void FixedUpdate() {
		if (Application.loadedLevelName == "Decision" && !changed) {
			StartCoroutine(Change());
			changed = true;
		}
		
		if (Application.loadedLevelName == "Celebration" && !changed) {
			GameObject finish = GameObject.FindGameObjectWithTag("Finish");
			var chooser = finish.GetComponent<WinnerChooser>();
			
			if (p1) {
				chooser.SetWinnerP1();
			} else {
				chooser.SetWinnerP2();
			}
			
			changed = true;
		}
	}
	
	public void Player1() {
		p1 = true;
	}
	
	public void Player2() {
		p1 = false;
	}
	
	IEnumerator Change() {
		yield return new WaitForSeconds(4);
		Application.LoadLevel("Celebration");
		
		changed = false;
	}
}
